src/test/fuzz/netbase_dns_lookup.cpp

```diff
@@ -12,27 +12,22 @@
 #include <string>
 #include <vector>
 
-namespace {
-FuzzedDataProvider* fuzzed_data_provider_ptr = nullptr;
-
-std::vector<CNetAddr> fuzzed_dns_lookup_function(const std::string& name, bool allow_lookup)
-{
-    std::vector<CNetAddr> resolved_addresses;
-    while (fuzzed_data_provider_ptr->ConsumeBool()) {
-        resolved_addresses.push_back(ConsumeNetAddr(*fuzzed_data_provider_ptr));
-    }
-    return resolved_addresses;
-}
-} // namespace
-
 FUZZ_TARGET(netbase_dns_lookup)
 {
     FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
-    fuzzed_data_provider_ptr = &fuzzed_data_provider;
     const std::string name = fuzzed_data_provider.ConsumeRandomLengthString(512);
     const unsigned int max_results = fuzzed_data_provider.ConsumeIntegral<unsigned int>();
     const bool allow_lookup = fuzzed_data_provider.ConsumeBool();
     const int default_port = fuzzed_data_provider.ConsumeIntegral<int>();
+
+    auto fuzzed_dns_lookup_function = [&](const std::string&, bool) {
```

Comment:
This is impossible to reuse if another `FUZZ_TARGET()` be added. It is ok since here we have just one `FUZZ_TARGET()` and if _ever_ another one is added and reuse needed, then it could be done in another way.

Reply:
I think (and hope) that the `netbase_dns_lookup` fuzzing harness should cover all our low-level DNS fuzzing needs, so I think the probability that we'll need to call `fuzzed_dns_lookup_function` from another `FUZZ_TARGET` in the future is relatively low.

In other words: let's cross that fuzzing bridge when we come to it :)

Reply:
YAGNI (you ain't gonna need it)